### PR TITLE
Start of library loader improvement

### PR DESCRIPTION
**Pull request by ghost**
This PR makes changes to the LibraryLoader class to add Try versions of the loader methods that return a boolean value for success and the pointer as an out param. 
This will make it easier for further changes when loading libraries without breaking changes.

### CODE REVIEW

File: src/Ultz.SuperInvoke.Loader/LibraryLoader.cs [hunk 1]
```diff
@@ -20,9 +20,28 @@ public abstract class LibraryLoader
         /// </summary>
         /// <param name="name">The name of the library to open.</param>
         /// <returns>The operating system handle for the shared library.</returns>
-        public IntPtr LoadNativeLibrary(string name)
+        public IntPtr LaodNativeLibrary(string name)
```
Comment (Perksey) [inline src/Ultz.SuperInvoke.Loader/LibraryLoader.cs]: Oh whoops there was a typo, one last thing before merge!

File: src/Ultz.SuperInvoke.Loader/LibraryLoader.cs [hunk 2]
```diff
@@ -20,9 +20,28 @@ public abstract class LibraryLoader
...
+
+        /// <summary>
+        ///     Loads a native library by name and returns an operating system handle to it.
+        /// </summary>
+        /// <param name="name">The name of the library to open.</param>
+        /// <param name="result">A pointer to the loaded library.</param>
+        /// <returns>The operating system handle for the shared library.</returns>
+        public bool TryLoadNativeLibrary(string name, out IntPtr result)
+        {
+            var success  = TryLoadNativeLibrary(new[] { name }, PathResolver.Default, out IntPtr libPtr);
```
Comment (Perksey) [inline src/Ultz.SuperInvoke.Loader/LibraryLoader.cs]: You likely don’t need to declare another variable for this, just pass result directly as out.

File: src/Ultz.SuperInvoke.Loader/LibraryLoader.cs [hunk 3]
```diff
@@ -69,18 +124,37 @@ public IntPtr LoadNativeLibrary(string[] names, PathResolver pathResolver)
...
+            result = IntPtr.Zero;
+
+            if (names == null || names.Length == 0)
+                return false;
+
             foreach (var name in names)
             {
-                ret = LoadWithResolver(name, pathResolver);
-                if (ret != IntPtr.Zero) break;
+                result = LoadWithResolver(name, pathResolver);
```
Comment (Perksey) [inline src/Ultz.SuperInvoke.Loader/LibraryLoader.cs:152]: This still uses the throwing method when it should use the try-style method (in its current form the first chance exceptions we sought to fix are still thrown!)

**Review by Perksey — CHANGES_REQUESTED** [review]

File: src/Ultz.SuperInvoke.Loader/LibraryLoader.cs (hunk 1 above)
Comment (KinsonDigital) [reply src/Ultz.SuperInvoke.Loader/LibraryLoader.cs:23]: Good catch!!
Was trying to slam it out before work started, hence the mistakes.  👍

Comment (KinsonDigital) [reply src/Ultz.SuperInvoke.Loader/LibraryLoader.cs]: fixed

File: src/Ultz.SuperInvoke.Loader/LibraryLoader.cs (hunk 2 above)
Comment (KinsonDigital) [reply src/Ultz.SuperInvoke.Loader/LibraryLoader.cs]: fixed

File: src/Ultz.SuperInvoke.Loader/LibraryLoader.cs (hunk 3 above)
Comment (KinsonDigital) [reply src/Ultz.SuperInvoke.Loader/LibraryLoader.cs:152]: Are you talking about the **TryLoadNativeLibrary** method on line 144?  I do not see any exceptions being thrown in there.

Comment (Perksey) [reply src/Ultz.SuperInvoke.Loader/LibraryLoader.cs:152]: Ah whoops, sorry I reviewed this on my phone late last night. This is all fine turns out.

**Review by Perksey — APPROVED** [review]

**Pull request merged**